src/GafferSceneUI/LightTool.cpp

```diff
@@ -760,495 +792,438 @@ class LightToolHandle : public Handle
+		// current contextexists. Returns false if no inspector or inspection result is available
+		// for `metaParameter`.
+		bool inspectionInfo( const InternedString &metaParameter, InspectionInfo &info ) const
 		{
-			if( !m_coneAngleInspector || ( m_handleType == HandleType::Penumbra && !m_penumbraAngleInspector ) )
+			Inspector::ResultPtr inspection = nullptr;
+			float originalValue = 0;
+
+			auto it = m_inspectors.find( metaParameter );
+			if( it == m_inspectors.end() )
 			{
 				return false;
 			}
 
-			// We can be called to check visibility for any scene location set in the current context, spot light
-			// or otherwise. If there isn't an inspection, this handle should be hidden (likely because the scene
-			// location is not a spot light).
+			if( inspection = it->second->inspect() )
+			{
+				auto originalData = runTimeCast<const FloatData>( inspection->value() );
+				assert( originalData );
+				originalValue = originalData->readable();
+				info = { inspection, originalValue };
+				return true;
+			}
 
-			Inspector::ResultPtr contextConeInspection = m_coneAngleInspector->inspect();
-			Inspector::ResultPtr contextPenumbraInspection = m_penumbraAngleInspector ? m_penumbraAngleInspector->inspect() : nullptr;
+			return false;
+		}
 
-			if( !contextConeInspection || ( m_handleType == HandleType::Penumbra && !contextPenumbraInspection ) )
+		bool allInspectionsEnabled() const
+		{
+			std::vector<const Inspector::Result *> requiredInspections = handleValueInspections();
+			if( requiredInspections.empty() )
 			{
 				return false;
 			}
 
-			// We are a spot light, but the penumbra will be hidden if it's too close to the cone angle, for
-			// the location we're attaching the handles to.
+			bool enabled = true;
+			for( const auto &i : requiredInspections )
+			{
+				enabled &= i && i->editable();
+			}
 
-			/// \todo This checks the penumbra / cone angles only for the last selected location, causing
-			/// repeated checks of the same location when `visible()` is called in a loop over multiple scene
-			/// locations. We rely on history caching to make this relatively fast, but ideally this could be
-			/// tested only once per selection list.
+			return enabled;
+		}
 
-			const auto &[coneInspection, coneAngle, penumbraInspection, penumbraAngle] = spotLightHandleAngles();
-			if( m_handleType == HandleType::Penumbra && penumbraAngle )
-			{
-				const float radius = m_visualiserScale * m_frustumScale * -10.f;
-				const V2f coneRaster = m_view->viewportGadget()->gadgetToRasterSpace(
-					V3f( 0, 0, radius ),
-					this
-				);
-				const M44f rot = M44f().rotate( V3f( 0, degreesToRadians( penumbraAngle.value() ), 0 ) );
-				const V2f penumbraRaster = m_view->viewportGadget()->gadgetToRasterSpace(
-					V3f( 0, 0, radius ) * rot,
-					this
-				);
+		// May be overriden to update internal state based on the `scenePath`
+		virtual void setScenePathInternal( ScenePathPtr scenePath )
```

Comment:
The function signature and naming here seems misleading - the function isn't responsible for setting the scene path, and `scenePath` is already available from `getScenePath()`. Perhaps `updateFromScenePath()` or `scenePathChanged()` would be more accurate?

Reply:
Right, fixed in https://github.com/GafferHQ/gaffer/pull/5483/commits/aee8ea294e5ee18d804f9b63fa2e5f8b57b6054f, but also modifed in https://github.com/GafferHQ/gaffer/commit/82e367704cc99ce474e7a68799b07d313d3ccfba.